cmd/uplink/cmd/setup.go

```diff
@@ -4,48 +4,41 @@
 package cmd
 
 import (
-	"crypto/rand"
 	"fmt"
 	"os"
 	"path/filepath"
 
-	base58 "github.com/jbenet/go-base58"
 	"github.com/spf13/cobra"
 	"go.uber.org/zap"
 
 	"storj.io/storj/internal/fpath"
 	"storj.io/storj/pkg/cfgstruct"
 	"storj.io/storj/pkg/miniogw"
 	"storj.io/storj/pkg/process"
-	"storj.io/storj/pkg/provider"
 )
 
+// UplinkFlags configuration flags
+type UplinkFlags struct {
+	APIKey        string `default:"" help:"the api key to use for the satellite" setup:"true"`
+	SatelliteAddr string `default:"localhost:7778" help:"the address to use for the satellite" setup:"true"`
+
+	miniogw.Config // TODO remove when GetMetainfo will be moved from minio
```

Comment:
Let's duplicate the code and remove `minio` dependency from uplink. With some comment like "keep in sync with etc" on both sides. But we can also do that in a separate PR.

Reply:
lets do this in separate PR